### PR TITLE
Unify troubleshoot specs

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What type of PR is this?

kind/chore

#### What this PR does / why we need it:

This PR is an attempt to streamline the process of writing new default collectors and analyzers by unifying the kots.io spec and the spec that is injected in go when generating a bundle through the KOTS UI. This code change makes it so that KOTS uses the kots.io spec in addition to the injected go spec and eliminates as much as possible from the go code.

#### Which issue(s) this PR fixes:

Fixes NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: pkg/supportbundle/spec.go [hunk 1]
```diff
@@ -174,8 +173,28 @@ func CreateRenderedSpec(appID string, sequence int64, kotsKinds *kotsutil.KotsKi
...
+		}
+	}
+
+	addDefaultTroubleshoot(supportBundle)
+	addDefaultDynamicTroubleshoot(supportBundle, app)
 	populateNamespaces(supportBundle, minimalRBACNamespaces)
+	if imageName != "" && pullSecret != nil {
+		populateImages(supportBundle, imageName, pullSecret)
+	}
+	deduplicatedCollectors(supportBundle)
```
Comment (sgalsaleh) [inline pkg/supportbundle/spec.go:197]: we should deduplicate analyzers as well if possible

Comment (emosbaugh) [reply pkg/supportbundle/spec.go:197]: thanks, updated

File: pkg/supportbundle/analyze.go [hunk 2]
```diff
@@ -1,130 +0,0 @@
...
-
-import (
-	"os"
-
-	"github.com/replicatedhq/kots/pkg/util"
-	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
-)
-
-func InjectDefaultAnalyzers(analyzers []*troubleshootv1beta2.Analyze) []*troubleshootv1beta2.Analyze {
-	analyzers = append(analyzers, getAPIReplicaAnalyzer())
```
Comment (emosbaugh) [inline pkg/supportbundle/analyze.go:11]: TODO: add getAPIReplicaAnalyzer back

**Review by sgalsaleh — APPROVED** [review]

**Comment by emosbaugh** [comment]
Below is the diff after this change:

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  creationTimestamp: null
  name: default-supportbundle
spec:
  analyzers: []
  collectors:
  - exec:
      args:
      - http://localhost:3030/goroutines
      collectorName: kotsadm-operator-goroutines
      command:
      - curl
      containerName: kotsadm-operator
      exclude: ""
      name: kots/admin_console
      namespace: default
      selector:
      - app=kotsadm-operator
      timeout: 10s
  - copyFromHost:
      collectorName: kurl-host-preflights
      exclude: ""
      extractArchive: true
      hostPath: /var/lib/kurl/host-preflights
      image: alpine
      imagePullPolicy: IfNotPresent
      name: kots/kurl/host-preflights
      namespace: ""
      timeout: 1m
  - configMap:
      collectorName: kurl-current-config
      exclude: ""
      includeAllData: true
      name: kurl-current-config
      namespace: kurl
  - configMap:
      collectorName: kurl-last-config
      exclude: ""
      includeAllData: true
      name: kurl-last-config
      namespace: kurl
  - collectd:
      collectorName: collectd
      exclude: ""
      hostPath: /var/lib/collectd/rrd
      image: alpine
      imagePullPolicy: IfNotPresent
      namespace: ""
      timeout: 5m
```

**Comment by emosbaugh** [comment]
Analyze screenshot

Screen Shot 2021-09-28 at 5 24 32 PM

File: pkg/supportbundle/defaultspec/spec.yaml [hunk 3]
```diff
@@ -95,6 +96,7 @@ spec:
         namespace: kurl
     - secret:
         collectorName: kotsadm-replicated-registry
+        name: kotsadm-replicated-registry # TODO: what is the correct name here?
```
Comment (divolgin) [inline pkg/supportbundle/defaultspec/spec.yaml:99]: this is the correct name for the secret used by kotsadm.  app secret names are generated dynamically

File: pkg/supportbundle/spec.go [hunk 4]
```diff
@@ -844,79 +696,22 @@ func getImageAndSecret(ctx context.Context, clientset kubernetes.Interface) (ima
...
-	analyzers = append(analyzers, &troubleshootv1beta2.Analyze{
-		Longhorn: &troubleshootv1beta2.LonghornAnalyze{
-			AnalyzeMeta: troubleshootv1beta2.AnalyzeMeta{
-				CheckName: "longhorn",
-			},
-		},
-	})
+	collects := []*troubleshootv1beta2.Collect{}
+	for _, collect := range next.Spec.Collectors {
+		if collect.Collectd != nil && collect.Collectd.Image == "alpine" { // TODO: is this too strong of an assumption?
```
Comment (divolgin) [inline pkg/supportbundle/spec.go:704]: I'm assuming the intention here is to override images in the embedded spec only.  If so, how about calling this before adding any other collectors.

Comment (emosbaugh) [reply pkg/supportbundle/spec.go:704]: good call. will make that change

File: pkg/supportbundle/defaultspec/spec.yaml (hunk 3 above)
Comment (emosbaugh) [reply pkg/supportbundle/defaultspec/spec.yaml]: i clarified this comment

**Review by divolgin — APPROVED** [review]

**Pull request merged**